static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -28,68 +28,112 @@ define([
+                            $header.css({ 'margin-top': state.adHeight });
+
+                            $adBanner.css({ 'max-height': state.adHeight });
+                            var userHasScrolledPastHeader = scrollY > headerHeight;
+                            if (userHasScrolledPastHeader) {
+                                $adBanner.css({
+                                    'position': 'absolute',
+                                    'top': headerHeight
+                                });
+                            } else {
+                                $adBanner.css({ 'position': 'fixed' });
+                            }
+
+                            var scrollIsAtTop = scrollY === 0;
+                            var diff = state.adHeight - state.previousAdHeight;
+                            var adHeightHasIncreased = diff > 0;
+                            if (!scrollIsAtTop && adHeightHasIncreased) {
+                                // If the user is not at the top and the ad height has increased,
+                                // we want to offset their scroll position
+                                window.scrollTo(scrollX, scrollY + diff);
+                            } else if (!state.firstRender) {
+                                // Otherwise we want to transition the change when it happens.
+                                // Avoid an initial transition when we apply the margin top for the first time
+                                $header.css('transition', 'margin-top 1s cubic-bezier(0, 0, 0, 0.985)');
+                            }
                         });
                     });
                 };
 
+                var update = (function () {
+                    var previousAdHeight;
+                    return function (options) {
+                        return getCachedAdHeight().then(function (adHeight) {
+                            return render({
+                                adHeight: adHeight,
+                                previousAdHeight: previousAdHeight || adHeight,
+                                firstRender: options.firstRender || false
+                            }).then(function () {
+                                previousAdHeight = adHeight;
+                            });
+                        });
+                    };
+                })();
+
                 //
                 // Side effects
                 //
 
-                mediator.on('window:throttledScroll', render);
-                render({ firstRender: true });
-                topAdRenderedPromise.then(render);
+                fastdom.write(function () {
+                    // will move into stylesheets when productionised
+                    $adBanner.css({'overflow': 'hidden', 'transition': 'max-height 1s cubic-bezier(0, 0, 0, 0.985)'});
+
+                    update({ firstRender: true }).then(function () {
+                        window.addEventListener('scroll', function () { update({}); });
```

Comment:
mediator `throttledScroll`? I'm impartial because we're already using fastdom and inside `update`